internal/transformer/classes/generators/dockerfilegenerators/winconsoleappdockerfilegenerator.go

```diff
@@ -0,0 +1,185 @@
+/*
+ *  Copyright IBM Corporation 2021
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package dockerfilegenerators
+
+import (
+	"encoding/xml"
+	"io/ioutil"
+	"net"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/konveyor/move2kube/environment"
+	"github.com/konveyor/move2kube/types/source/dotnet"
+	transformertypes "github.com/konveyor/move2kube/types/transformer"
+	"github.com/konveyor/move2kube/types/transformer/artifacts"
+	"github.com/sirupsen/logrus"
+)
+
+// WinConsoleAppDockerfileGenerator implements the Transformer interface
+type WinConsoleAppDockerfileGenerator struct {
+	Config transformertypes.Transformer
+	Env    *environment.Environment
+}
+
+// Init Initializes the transformer
+func (t *WinConsoleAppDockerfileGenerator) Init(tc transformertypes.Transformer, env *environment.Environment) (err error) {
+	t.Config = tc
+	t.Env = env
+	return nil
+}
+
+// GetConfig returns the transformer config
+func (t *WinConsoleAppDockerfileGenerator) GetConfig() (transformertypes.Transformer, *environment.Environment) {
+	return t.Config, t.Env
+}
+
+// BaseDirectoryDetect runs detect in base directory
+func (t *WinConsoleAppDockerfileGenerator) BaseDirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	return nil, nil, nil
+}
+
+// parseAppConfig parses the app.config file for ports
+func (t *WinConsoleAppDockerfileGenerator) parseAppConfig(baseDir string) []string {
+	appConfigFile, err := os.Open(filepath.Join(baseDir, "App.config"))
```

Comment:
Won't it be better to return error in this case

Reply:
I have logged the error.

Reply:
When the function is not able to perform the primary job it is supposed to do, then it is better to return error, and the caller has to decide how to proceed further.  Since in this function it is about parsing a config file, and the function is not even able to open the file, it is better to return error and log/handle it in the caller code.

Reply:
Fixed